flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecHashJoin.java

```diff
@@ -111,6 +111,9 @@ public class BatchExecHashJoin extends ExecNodeBase<RowData>
     @JsonProperty(FIELD_NAME_ESTIMATED_RIGHT_ROW_COUNT)
     private final long estimatedRightRowCount;
 
+    @JsonProperty(FIELD_NAME_IS_JOIN_HINT)
+    private final boolean withHint;
```

Comment:
It will be assigned a default value of false, but complied plan will no longer undergo processors optimization to convert to a AdaptiveJoin, so it will not be affected.